server/status: add uptime to load related metrics

darinpp opened the pull request:
The load related metrics are exported under _status/load and contain absolute wall time as well as cumulative CPU time consumed by the process (user and system). These are used to compute current CPU usage percentage.
To do that however, a client needs to do two gets - one to get the baseline cumulative CPU time together with wall time and then a bit later another get to obtain the changed values. The CPU usage percentage then is computed as the delta in the CPU time divided by the delta in wall time. The two required gets delay the computation of the initial CPU usage. This PR adds uptime to the exported metrics which allows computing CPU usage with just a single get. With just one get, a client can simply divide the CPU time by the uptime and determine the CPU usage percentage since process start.
The PR exports the uptime as seconds for compatibility with the normal metric but takes use of the fractional portion to allow sub-second precision.
It also changes the reporting of CPU time to be since the start of the server instead of start of the process. This makes the CPU time a better match to the uptime value and solves the problem where the server process may have been started by an already running process (sqlstarter for example) via calling exec.

Part of https://cockroachlabs.atlassian.net/browse/CC-8504

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/102665)
<!-- Reviewable:end -->

darinpp committed:
server/status: add uptime to load related metrics

The load related metrics are exported under _status/load and contain
absolute wall time as well as cumulative CPU time consumed by the
process (user and system). These are used to compute current CPU usage
percentage.
To do that however, a client needs to do two gets - one to get the
baseline cumulative CPU time together with wall time and then a bit
later another get to obtain the changed values. The CPU usage percentage
then is computed as the delta in the CPU time divided by the delta in
wall time. The two required gets delay the computation of the initial
CPU usage. This PR adds uptime to the exported metrics which allows
computing CPU usage with just a single get. With just one get, a client
can simply divide the CPU time by the uptime and determine the CPU usage
percentage since process start.
The PR exports the uptime as seconds for compatibility with the normal
metric but takes use of the fractional portion to allow sub-second
precision.
It also changes the reporting of CPU time to be since the start of the
server instead of start of the process. This makes the CPU time a better
match to the uptime value and solves the problem where the server
process may have been started by an already running process (sqlstarter
for example) via calling exec.

Part of https://cockroachlabs.atlassian.net/browse/CC-8504

Release note: None

darinpp commented:
@JeffSwenson Setting the baseline values for uptime and CPU usage is now part of the construction of the handler.

JeffSwenson commented on pkg/server/load_endpoint.go:
```diff
@@ -16,65 +16,114 @@ import (
+) (*loadEndpoint, error) {
+	initUserTimeMillis, initSysTimeMillis, err := status.GetProcCPUTime(context.Background())
+	if err != nil {
+		return nil, err
```
nit: add an error.Wrap annotation to this error so that in the unlikely event it occurs in production we get a stack trace that points to the load endpoint initialization.

jeffswenson reviewed (approved):
LGTM

Nice work!

darinpp commented:
bors r+
